test/studies/IMSuite/leader_elect_lcr/leader_elect_lcr.chpl

```diff
@@ -0,0 +1,194 @@
+ /* This file is part of IMSuite Benchamark Suite.
+
+    This file is licensed to You under the Eclipse Public License (EPL);
+    You may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.opensource.org/licenses/eclipse-1.0.php
+
+   (C) Copyright IMSuite 2013-present.
+ */
+
+ /*
+    leader_elect_lcr aims to elect a leader from a set of nodes,
+    on the basis of leader election algorithm by Le Lang et al..
+    The algorithm is aimed towards unidirectional ring networks.
+
+    @author Suyash Gupta
+    @author V Krishna Nandivada
+
+    Ported By: Kushal Singh
+ */
+
+ module leader_elect_lcr {
+
+     config const inputFile = "input/inputleader_elect_lcr_8.txt";
+     config const outputFile = "outputleader_elect_lcr_8.txt";
+
+     /* Parameter to enable verification of the output */
+     config const verify = false;
+
+     /* Stores the total number of process */
+     var processes: int;
+     /* Represents the domain over the number of nodes */
+     var D : domain(1);
+     /* Stores the id's of each of the nodes in the ring */
+     var IdStore: [D] int;
+
+     /* Parameters to enable execution with load */
+     config const loadValue=0;
+     var nval:[D] int;
+
+    /*
+        ```Process``` specifies the structure for each abstract node
+        part of the Leader election algorithm.
+    */
+    class Process {
+        /* Specifies the identifier for each node. */
+        var id: int;
+        /* Represents the identifier that the process will send to its neighbor. */
+        var send: int;
+        /* Represents the identifier of the leader. */
+        var leaderId: int;
+        /* Represents the identifier that the process receives from its neighbor. */
+        var receivedId: int;
+        /* Represents the status of the process, set to true for leader. */
+        var status: bool;
+     }
+
+     /* Abstract node representation */
+     var processSet: [D] Process;
+
+    /*
+        Acts as the starting point for the program execution.
+        ```main``` performs the task of accepting the input from the user
+        specified file, electing the leader and transmitting information,
+        printing the output and validating the result.
+    */
+     proc main(){
+
+        var inFile = open(inputFile, iomode.r);
+        var reader = inFile.reader();
+        processes = reader.read(int);
+        D = {0..(processes-1)};
+
+        for i in 0..(processes-1) do IdStore[i]=reader.read(int);
+
+        reader.close();
+        inFile.close();
+
+        initialize();
+        for round in 0..processes-1 do leader_elect();
```

Comment:
You don't need to make this change, just mentioning a Chapel feature:

You can write this more cleanly as:
```chpl
for round in 0..#processes do leader_elect();
```

'low..#N' will result in a range of size N starting at low. For example, "5..#10" will be the range "5..14". It can save a lot of "-1" and parens.